sdk/daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala

```diff
@@ -1139,14 +1141,130 @@ private[lf] object SBuiltin {
     ): Control[Nothing] = {
       def coid = getSContractId(args, 0)
       val (actualTmplId, record) = getSAnyContract(args, 1)
-      if (!interfaceInstanceExists(machine, ifaceId, actualTmplId)) {
-        Control.Error(IE.ContractDoesNotImplementInterface(ifaceId, coid, actualTmplId))
-      } else {
-        Control.Value(record)
+      castAnyInterface(machine, ifaceId, coid, actualTmplId, record)(Control.Value)
+    }
+  }
+
+  private[this] def castAnyInterface[Q](
+      machine: Machine[_],
+      ifaceId: TypeConName,
+      coid: V.ContractId,
+      actualTmplId: TypeConName,
+      record: SRecord,
+  )(k: SValue => Control[Q]): Control[Q] =
+    if (!interfaceInstanceExists(machine, ifaceId, actualTmplId)) {
+      Control.Error(IE.ContractDoesNotImplementInterface(ifaceId, coid, actualTmplId))
+    } else {
+      k(record)
+    }
+
+  /** Fetches the requested contract Id, upgrades it to the preferred template version for the same package name,
+    * and compares the computed views according to the old and the new versions. If the two views agree then caches
+    * the upgraded contract, inserts a Fetch node, and returns the upgraded contract.
+    */
+  final case class SBFetchInterface(interfaceId: TypeConName) extends UpdateBuiltin(1) {
+    override protected def executeUpdate(
+        args: util.ArrayList[SValue],
+        machine: UpdateMachine,
+    ): Control[Question.Update] = {
+      val coid = getSContractId(args, 0)
+      fetchAny(machine, None, coid, SValue.SValue.None) { (maybePkgName, sourceContract) =>
+        maybePkgName match {
+          case None =>
+            crash(s"unexpected contract instance without packageName")
```

Comment:
crashes should be limited to unexpected error. 
What it is the case of this one ?

Reply:
As discussed, this comment was about 
```
 crash(s"failed to convert contract $coid to type $dstTmplId")
 ```
below, which I have replaced with a WronglyTyped error.